### PR TITLE
[Pipeline](select node) Support select node on pipeline engine

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.92 seconds
 load time: 431 seconds
 storage size: 17123356366 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221208070827_clickbench_pr_60020.html

File: be/src/pipeline/exec/operator.h [hunk 1]
```diff
@@ -318,6 +319,10 @@ class Operator : public OperatorBase {
     virtual Status get_block(RuntimeState* state, vectorized::Block* block,
                              SourceState& source_state) override {
         SCOPED_TIMER(_runtime_profile->total_time_counter());
+        if (!is_source()) {
```
Comment (HappenLee) [inline be/src/pipeline/exec/operator.h:322]: ？what the work do here， should get block from `_node` ?

File: be/src/pipeline/exec/operator.h [hunk 2]
```diff
@@ -338,22 +346,51 @@ class Operator : public OperatorBase {
...
+    Status get_block(RuntimeState* state, vectorized::Block* block,
+                     SourceState& source_state) override {
+        SCOPED_TIMER(_runtime_profile->total_time_counter());
+        bool eos = false;
+        RETURN_IF_ERROR(_node->pull(state, block, &eos));
+        source_state = eos ? SourceState::FINISHED : SourceState::DEPEND_ON_SOURCE;
+        return Status::OK();
+    }
+
+    Status finalize(RuntimeState* state) override { return Status::OK(); }
```
Comment (HappenLee) [inline be/src/pipeline/exec/operator.h:370]: Do not need override the method `finalize` and `can_read`

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**